plugins/thunderbolt/fu-thunderbolt-image.c

```diff
@@ -520,11 +573,9 @@ get_device_locations (guint16 id, const FuThunderboltFwObject *fw, GError **erro
 	case 0x15EA:
 	case 0x15EF:
 		/* if the controller has multi controller entry need to
-		 * compare it against the image. */
-		if (read_drom_entry_location (fw, DROM_ENTRY_MC, &TR[1],
-					      &error_local))
+		 * compare it against the image first. */
+		if (compare_device_mc (fw, image, error))
```

Comment:
You need the error_local and g_deubg perhaps? Either you return VALIDATION_FAILED and set error, or return TR and don't touch error. If that's what's happening, it's slightly odd to rely on the fallthrough IMHO.

Reply:
Hmm, so the idea is to return VALIDATION_FAILED and propagate error if the `compare_device_mc ()` fails. Otherwise return TR. I wonder if this would be better than the fallthrough case:
```
if (compare_device_mc (fw, image, error))
        return TR;
return NULL;
```
?